content/docs/state-and-lifecycle.md

```diff
@@ -300,84 +300,84 @@ ReactDOM.render(
+२) React तब `Clock` कम्पोनेंटको `render()` मेथड कल गर्दछ। Reactले स्क्रिनमा प्रदर्शन हुने कुरा यसरी सिक्छ। `Clock`को रेन्डर आउटपुटसँग मेल गर्न Reactले तब DOM अपडेट गर्दछ।
 
-3) When the `Clock` output is inserted in the DOM, React calls the `componentDidMount()` lifecycle method. Inside it, the `Clock` component asks the browser to set up a timer to call the component's `tick()` method once a second.
+३) जब `Clock` आउटपुट DOM मा सम्मिलित हुन्छ, Reactले `componentDidMount()` लाइफसाइकल विधिलाई कल गर्दछ। यस भित्र `Clock` कम्पोनेंटले सेकेन्डको एक पटक कम्पोनेंटको `tick()` विधि कल गर्न ब्राउजरलाई टाइमर सेट अप गर्न सोध्दछ।
 
-4) Every second the browser calls the `tick()` method. Inside it, the `Clock` component schedules a UI update by calling `setState()` with an object containing the current time. Thanks to the `setState()` call, React knows the state has changed, and calls the `render()` method again to learn what should be on the screen. This time, `this.state.date` in the `render()` method will be different, and so the render output will include the updated time. React updates the DOM accordingly.
+४) प्रत्येक सेकेन्ड ब्राउजरले `tick()` विधि कल गर्दछ। यसको भित्र `Clock` कम्पोनेंटले हालको समय समावेश गरेको objectसँग `setState()` कल गरेर UI अपडेट अनुसूची गर्छ। यो `setState()` कलको कारण Reactलाई थाहा छ state परिवर्तन भएको छ र स्क्रिनमा के हुनुपर्दछ भनेर जान्न उसले फेरि `render()` विधिलाई कल गर्दछ। यस पटक `this.state.date`को `render()` विधि फरक हुनेछ र त्यसैले रेन्डर आउटपुटमा अपडेट गरिएको समय समावेश हुनेछ। React तदनुसार DOM अपडेट गर्दछ।
 
-5) If the `Clock` component is ever removed from the DOM, React calls the `componentWillUnmount()` lifecycle method so the timer is stopped.
+५) यदि `Clock` कम्पोनेंट कहिले पनि DOM बाट हटाइन्छ, Reactले `componentWillUnmount()` लाइफसाइकल विधि कल गर्दछ त्यसैले टाइमर रोकिन्छ।
 
-## Using State Correctly {#using-state-correctly}
+## कसरी Stateको सही प्रयोग गर्ने {#using-state-correctly}
 
-There are three things you should know about `setState()`.
+त्यहाँ तीनवटा कुरा छन् जुन तपाईले `setState()`को बारेमा जान्नुपर्दछ।
 
-### Do Not Modify State Directly {#do-not-modify-state-directly}
+### Stateको सिधा परिमार्जन नगर्नुहोस् {#do-not-modify-state-directly}
 
-For example, this will not re-render a component:
+उदाहरण को लागी, यसले कम्पोनेंट पुन: रेंडर गर्दैन:
 
 ```js
-// Wrong
+// गलत
 this.state.comment = 'Hello';
 ```
 
-Instead, use `setState()`:
+यसको सट्टा `setState()` प्रयोग गर्नुहोस्
 
 ```js
-// Correct
+// सही
 this.setState({comment: 'Hello'});
 ```
 
-The only place where you can assign `this.state` is the constructor.
+कन्स्ट्रक्टर एक मात्र स्थान जहाँ तपाईं `this.state`मा मान प्रदान गर्न सक्नुहुन्छ।
 
-### State Updates May Be Asynchronous {#state-updates-may-be-asynchronous}
+### State अपडेटहरू एसिन्क्रोनस हुन सक्छ {#state-updates-may-be-asynchronous}
```

Comment:
I would not translate asynchronous rather than transliterating it (Feel free to translate it to something meaningful?)